### PR TITLE
Add Jenkins profile to use Maven toolchains

### PR DESCRIPTION
**Pull request by nacx**
This is a PR to experiment with Maven toolchains approach to let Jenkins use the right JDK to build jclouds.
Do not merge yet.

### CODE REVIEW

**Comment by nacx** [comment]
rebuild please

**Comment by nacx** [comment]
rebuild please

**Comment by nacx** [comment]
rebuild please

**Comment by nacx** [comment]
rebuild please

**Comment by nacx** [comment]
rebuild please

**Comment by nacx** [comment]
rebuild please

File: project/pom.xml [hunk 1]
```diff
@@ -1375,5 +1375,32 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>jenkins</id>
```
Comment (demobox) [inline project/pom.xml]: @nacx Just out of curiosity: is the `jenkins` profile enabled automagically, or do we need to do that in the job configuration somewhere?

**Comment by nacx** [comment]
Back to green. Thanks @demobox!
WDYT about the toolchain approach? Shall we go for it?
@gaul are the Tier fixes OK?

**Comment by demobox** [comment]
> WDYT about the toolchain approach? Shall we go for it?

It seems the least invasive solution both in terms of changes to the code and the Jenkins configuration - I wonder if there's a way to make the required changes on the Jenkins side via API calls, actually?

I don't know enough about the various plugins involved to have a strong opinion, though - perhaps @gaul or @abayer have more thoughts on those?

As regards the `tier` changes, I think we should probably pull them out into a separate PR and clean them up so they really do The Right Thing (rather than just fixing the build). I expect Andrew G will be best placed to do that, though.

**Comment by gaul** [comment]
@nacx Sorry I missed this earlier.  I fixed the tiering regression in 5448b92ee8c9ba230df9ad8e0d89b070f5b869ae.

**Comment by nacx** [comment]
Ok. I think we're OK with the toolchains approach. I tried to script the scripts to bulk-update all the jobs in Jenkins but I got permission errors; I don't have permissions to run scripts and did not found where to configure that, so I've manually updated all HEAD, 2.0.x, live test and pull request jobs.

I'm merging this now to make the builds happy again, although a local build, with the blobstore changes in this PR reverted and rebased to the latest master version still presented some test failures. @gaul would you mind having a look at those?

**Comment by demobox** [comment]
> so I've manually updated all HEAD, 2.0.x, live test and pull request jobs

Egads - sorry the automated approach didn't work :-( Many thanks for going through all that manually, @nacx!

**Comment by nacx** [comment]
I added one additional profile, jenkins-jdk8 to preconfigure the JDK8 toolchain. It's needed by how we have configured the "jclouds-compat" build, but I'm pretty sure we can clean that by using variables to define the jdk version and vendor instead of having two profiles. I'll have a look at that soon, but I wanted to bring things back to a stable status furst.

**Comment by gaul** [comment]
> I'm merging this now to make the builds happy again, although a local build, with the blobstore changes in this PR reverted and rebased to the latest master version still presented some test failures. @gaul would you mind having a look at those?

Can you be more specific about these failures?  I just ran `mvn clean verify` against e0e3519ed8397aa5697679f383cfdad5604480b3 without errors.

**Comment by gaul** [comment]
I see the error on Jenkins -- I was not running the labs integration tests.  Fixed in jclouds/jclouds-labs@adb4087ab1c37bf8435b293778767e312d89e18b.

**Comment by nacx** [comment]
And looks like everything is back to green! Thanks @gaul!

It looks like failures only happen on my machine (but consistently). Is it possible that they fail on Mac? I recall some tests failing in the past because it did not support file attributes. Is the tier information stored there? That could explain the failures.

This is my environment:

```bash
nacx@maqui:~/src/asf/jclouds (master) $ mvn -version
Apache Maven 3.2.3 (33f8c3e1027c3ddde99d3cdebad2656a31e8fdf4; 2014-08-11T22:58:10+02:00)
Maven home: /opt/maven
Java version: 1.7.0_51, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.7.0_51.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.12.6", arch: "x86_64", family: "mac"
```

And here are the test failures.

```bash
nacx@maqui:~/src/asf/jclouds (master) $ cat /Users/nacx/src/asf/jclouds/apis/filesystem/target/surefire-reports/TestSuite.txt
-------------------------------------------------------------------------------
Test set: TestSuite
-------------------------------------------------------------------------------
Tests run: 104, Failures: 4, Errors: 0, Skipped: 9, Time elapsed: 9.479 sec <<< FAILURE! - in TestSuite
testPutBlobTierArchive(org.jclouds.filesystem.integration.FilesystemBlobIntegrationTest)  Time elapsed: 0.007 sec  <<< FAILURE!
java.lang.AssertionError:
Expecting:
 <STANDARD>
to be equal to:
 <ARCHIVE>
but was not.
	at org.jclouds.blobstore.integration.internal.BaseBlobIntegrationTest.testPutBlobTierHelper(BaseBlobIntegrationTest.java:790)
	at org.jclouds.blobstore.integration.internal.BaseBlobIntegrationTest.testPutBlobTierArchive(BaseBlobIntegrationTest.java:760)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:696)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.access$000(SuiteRunner.java:38)
	at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:382)
	at org.testng.internal.thread.ThreadUtil$2.call(ThreadUtil.java:64)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)

testPutBlobTierArchiveMultipart(org.jclouds.filesystem.integration.FilesystemBlobIntegrationTest)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError:
Expecting:
 <STANDARD>
to be equal to:
 <ARCHIVE>
but was not.
	at org.jclouds.blobstore.integration.internal.BaseBlobIntegrationTest.testPutBlobTierHelper(BaseBlobIntegrationTest.java:790)
	at org.jclouds.blobstore.integration.internal.BaseBlobIntegrationTest.testPutBlobTierArchiveMultipart(BaseBlobIntegrationTest.java:775)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:696)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.access$000(SuiteRunner.java:38)
	at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:382)
	at org.testng.internal.thread.ThreadUtil$2.call(ThreadUtil.java:64)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)

testPutBlobTierInfrequent(org.jclouds.filesystem.integration.FilesystemBlobIntegrationTest)  Time elapsed: 0.009 sec  <<< FAILURE!
java.lang.AssertionError:
Expecting:
 <STANDARD>
to be equal to:
 <INFREQUENT>
but was not.
	at org.jclouds.blobstore.integration.internal.BaseBlobIntegrationTest.testPutBlobTierHelper(BaseBlobIntegrationTest.java:790)
	at org.jclouds.blobstore.integration.internal.BaseBlobIntegrationTest.testPutBlobTierInfrequent(BaseBlobIntegrationTest.java:755)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:696)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.access$000(SuiteRunner.java:38)
	at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:382)
	at org.testng.internal.thread.ThreadUtil$2.call(ThreadUtil.java:64)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)

testPutBlobTierInfrequentMultipart(org.jclouds.filesystem.integration.FilesystemBlobIntegrationTest)  Time elapsed: 0.009 sec  <<< FAILURE!
java.lang.AssertionError:
Expecting:
 <STANDARD>
to be equal to:
 <INFREQUENT>
but was not.
	at org.jclouds.blobstore.integration.internal.BaseBlobIntegrationTest.testPutBlobTierHelper(BaseBlobIntegrationTest.java:790)
	at org.jclouds.blobstore.integration.internal.BaseBlobIntegrationTest.testPutBlobTierInfrequentMultipart(BaseBlobIntegrationTest.java:770)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:696)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.access$000(SuiteRunner.java:38)
	at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:382)
	at org.testng.internal.thread.ThreadUtil$2.call(ThreadUtil.java:64)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)
```

**Comment by gaul** [comment]
@nacx You have described the problem exactly -- Java on Mac OS does not support extended attributes.  Could you add some test exclusions for tiers similar to the ones we already have for other attributes?

**Comment by nacx** [comment]
Sure I'll do. I was just guessing and asking before diving into the code :) I'll open a PR in short. Thanks for confirming!

**Comment by nacx** [comment]
Done: https://github.com/jclouds/jclouds/pull/1150

**Pull request merged**